frontend/src/pages/eventoverview.js

```diff
@@ -0,0 +1,174 @@
+import React from 'react'
+import Router from 'next/router'
+import {
+  Table,
+  Row,
+  Button,
+  Col,
+  Form,
+  FormGroup,
+  Label,
+  Input,
+  Container,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter
+} from 'reactstrap'
+import ActionButton from '../components/actionButton'
+import Nav from '../components/nav'
+import { getAllEvents, createEvent } from '../utils/api'
+
+class EventOverview extends React.Component<Props> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      events: [],
+      modal: false,
+      name: '',
+      date: '',
+      startTime: 0,
+      endTime: 0,
+      location: '',
+      description: ''
+    }
+  }
+
+  async componentDidMount() {
+    const events = await getAllEvents()
+    if (events) {
+      this.setState({
+        events: events.result
+      })
+    }
+  }
+
+  openModal = () => {
+    this.setState({ modal: true })
+  }
+
+  closeModal = () => {
+    this.setState({ modal: false })
+  }
```

Comment:
You should be able to consolidate these two functions (openModal, closeModal) into one function that just changes the modal state with !modal. This way its more concise with just one function that opens and closes the modal and is still readable

Reply:
That way also the modal will be closed by default on the page and then the only way to open the modal is through one certain button and the only way to close the modal is through another button or two on the modal itself, so the modal state should work out nicely